### PR TITLE
Prevent error with orphaned shares when updating user mount cache

### PR DESCRIPTION
**Pull request by PVince81**
Forward port of https://github.com/owncloud/core/pull/25948 to 9.2

@jvillafanez @VicDeo @DeepDiver1975 @butonic

### CODE REVIEW

File: apps/files_sharing/lib/SharedMount.php [hunk 1]
```diff
@@ -241,6 +241,11 @@ public function getStorageNumericId() {
 			->from('filecache')
 			->where($query->expr()->eq('fileid', $query->createNamedParameter($this->getStorageRootId())));
 
-		return $query->execute()->fetchColumn();
+		$result = $query->execute();
+		$row = $result->fetch();
```
Comment (DeepDiver1975) [inline apps/files_sharing/lib/SharedMount.php]: close cursor?

Comment (PVince81) [inline apps/files_sharing/lib/SharedMount.php]: will do

**Comment by jvillafanez** [comment]
Besides the comment, 👍

**Comment by PVince81** [comment]
backport for the closeCursor commit, to make sure the stable9.1 version is the same: https://github.com/owncloud/core/pull/25979

**Pull request merged**